### ISSUE TITLE
`scan-files` does not work

### ISSUE THREAD

**Issue by ghost** [maglnet/ComposerRequireChecker#542]
Composer Require Checker version: 4.7.1

I have symfony app and `bin/console` file with a symbol from package that doesn't exist in root composer.json. This symbol is not reported by this tool (it is when used inside a class in src/ dir)

My `composer-require-checker-.json`:
```json
{
  "symbol-whitelist" : [
  ],
  "php-core-extensions" : [
    "Core",
    "date",
    "json",
    "hash",
    "pcre",
    "Phar",
    "Reflection",
    "SPL",
    "random",
    "standard"
  ],
  "scan-files" : [
    "bin/console"
  ]
}
```

**Comment by Ocramius** [comment]
Could you perhaps provide a snippet / which symbol?

**Comment by ghost** [comment]
I've added `new SomeClassFromSoftDep()` inside `bin/console` and executed this tool. Then I added the same line to random class in `src/`, executed this tool and it was reported. 
However I suspect that option `scan-files` is used often so someone should have already reported this since Sep 27, 2023. I will try to create a reproducer.

**Comment by Ocramius** [comment]
Could you check whether the path is correct at https://github.com/maglnet/ComposerRequireChecker/blob/1763ae438e83478fa11390e85d29d5ffbaa9e326/src/ComposerRequireChecker/Cli/CheckCommand.php#L161 ? 

I think it expects a `glob` pattern there, so I'm not sure if it's finding `bin/console` :thinking:

**Comment by ghost** [comment]
> Could you check whether the path is correct at

I'm using a phar file. Not sure if I can debug that.

Here is a reproducer: https://github.com/javaDeveloperKid/reproducer2406
![image](https://github.com/maglnet/ComposerRequireChecker/assets/25783196/8ec0dc65-0b95-42fe-9fee-2316b85af7fe)

**Comment by ghost** [comment]
When I rename to `bin/console.php` and set `scan-files` to `bin/*.php` is doesn't work either.

**Comment by Ocramius** [comment]
can already see that something isn't right:

```
Run composer recipes at any time to see the status of your Symfony recipes.

Executing script cache:clear [KO]
 [KO]
Script cache:clear returned with error code 255
!!  
!!  Fatal error: Uncaught Error: Class "Symfony\Component\Routing\Route" not found in /app/bin/console:15
!!  Stack trace:
!!  #0 {main}
!!    thrown in /app/bin/console on line 15
!!  
Script @auto-scripts was called via post-install-cmd
```

**Comment by ghost** [comment]
yeah, This line in bin/console should be 2 lines lower. Fixed.

**Comment by Ocramius** [comment]
`.phar` strikes again.

I did a fresh install:

```
mkdir tools/
echo '{"require":{"maglnet/composer-require-checker": "^4.11"}}' > tools/composer.json
docker run -ti --rm -v $(pwd)/tools:/app -w /app composer:2.7.7 composer install
docker run -ti --rm -v $(pwd):/app -w /app composer:2.7.7 composer install
docker run -ti --rm -v $(pwd):/app -w /app php:8.3.8-cli ./tools/vendor/bin/composer-require-checker
```

and it produced

```
root@8571683c8562:/app# ./tools/vendor/bin/composer-require-checker 
ComposerRequireChecker 4.11.0@c6c555e799bee50810fd84933ca1f0b276379ccf
The following 2 unknown symbols were found:
+-------------------------------------+--------------------+
| Unknown Symbol                      | Guessed Dependency |
+-------------------------------------+--------------------+
| Symfony\Component\HttpKernel\Kernel |                    |
| Symfony\Component\Routing\Route     |                    |
+-------------------------------------+--------------------+
```

@maglnet I add this to my list of "stop using `.phar`s" :D

**Comment by ghost** [comment]
FYI: I use 4.7.1 because >=4.8.0 dropped support for PHP 8.1 (my environment php version)

**Comment by Ocramius** [comment]
> (my environment php version)

Shouldn't matter for running this tool: you can run this tool with any newer PHP version, and it should still scan older code just fine, as long as it is still parsed (example: PHP4 syntax may be a problem)

**Comment by ghost** [comment]
Ok, 4.11 works.

`docker run -ti --rm -v $(pwd):/app -w /app php:8.3.8-cli php composer-require-checker-411.phar`

**Comment by Ocramius** [comment]
Closing then :-) 

I think this is related to #511

**Comment by ghost** [comment]
Can I ask how do I deal with `Extension "pcntl" does not exist` error when running this tool via docker image?
I tried 4.11 with php:8.3.8-cli image on the real project but it reports errors as above.

**Comment by Ocramius** [comment]
could install `pcntl` via something like https://github.com/mlocati/docker-php-extension-installer/blob/7ccc11b3760b170ecec6b5e0b55b7465b6dfb332/README.md
